server/tests/personality.e2e.spec.ts

```diff
@@ -110,18 +266,129 @@ describe("PersonalityController (e2e)", () => {
             .expect(200);
     });
 
+    /**
+     * Test: DELETE /api/personality/:id - Delete Personality
+     * 
+     * Purpose: Validates personality deletion functionality with soft-delete behavior
+     * Business Logic:
+     * - Implements soft delete (sets isDeleted flag rather than removing record)
+     * - Preserves personality data for audit trails and referential integrity
+     * - Removes personality from all public API responses
+     * - Maintains data consistency for existing claims/reviews that reference this personality
+     * 
+     * Validates:
+     * - HTTP 200 response for successful deletion
+     * - Personality becomes inaccessible via public APIs
+     * - Soft delete implementation preserves data integrity
+     */
     it("/api/personality/:id (DELETE) - Delete Personality", () => {
         return request(app.getHttpServer())
             .delete(`/api/personality/${personalityId}`)
             .expect(200);
     });
 
+    /**
+     * Test: GET /api/personality/:id - Verify Deleted Personality Inaccessibility
+     * 
+     * Purpose: Validates that soft-deleted personalities are not accessible via public API
+     * Business Logic:
+     * - Deleted personalities should not return meaningful data
+     * - API maintains consistent response format while hiding deleted content
+     * - Confirms soft delete implementation works correctly
+     * - Prevents data exposure after deletion
+     * 
+     * Validates:
+     * - HTTP 200 response (endpoint handles deleted resources gracefully)
+     * - Response body is empty object {} (indicating personality is no longer accessible)
+     * - Soft delete behavior prevents data exposure after deletion
+     * - API consistency in handling deleted resources
+     */
     it("/api/personality/:id (GET) - Get by id", () => {
         return request(app.getHttpServer())
             .get(`/api/personality/${personalityId}`)
             .query({ language: "pt" })
             .expect(200)
-            .expect(({ body }) => expect(body).toMatchObject({}));
+            .expect(({ body }) => {
+                expect(body).toMatchObject({});
+                expect(Object.keys(body).length).toBe(0);
+            });
+    });
+
+    /**
+     * Test: GET /api/personality/:id - Handle Non-existent Personality ID
+     * 
+     * Purpose: Validates API behavior when requesting non-existent personality
+     * Business Logic:
+     * - Should handle requests for non-existent personalities gracefully
+     * - Returns empty object instead of throwing errors
+     * - Maintains consistent API response format
+     * 
+     * Validates:
+     * - HTTP 200 response (not 404)
+     * - Response body is empty object {}
+     * - API handles invalid ObjectId gracefully
+     */
+    it("/api/personality/:id (GET) - Should handle non-existent personality ID", () => {
+        const { ObjectId } = require("mongodb");
```

Comment:
issue: make this consistent with places using mongoose Types and this should be declared at the top level of the file. Review calls like this in the entire PR.